shotover-proxy/src/codec/cassandra.rs

```diff
@@ -196,6 +205,26 @@ fn reject_protocol_version(version: u8) -> CodecReadError {
     ))])
 }
 
+fn reject_compression(compression: &String) -> CodecReadError {
+    info!(
+        "Rejecting compression option {} (configure the client to use no compression)",
+        compression
+    );
+
+    CodecReadError::RespondAndThenCloseConnection(vec![Message::from_frame(Frame::Cassandra(
+        CassandraFrame {
+            version: Version::V4,
+            stream_id: 0,
```

Comment:
can we take this from the incoming message? The spec makes no guarantee that the first message has stream_id 0